sonar-pmd-lib/src/main/java/org/sonar/plugins/pmd/rule/RulesDefinitionXmlLoader.java

```diff
@@ -380,22 +381,43 @@ private static void buildRule(RulesDefinition.NewRepository repo, String key, St
         }
     }
 
-    private static String[] filterNonSonarTags(List<String> tags) {
+    private static String[] determineSonarRuleTags(String name, List<String> inputTags) {
+        List<String> ruleTags = new ArrayList(inputTags);
+        boolean hasTagTests = ruleTags.contains(TAG_TEST_SOURCES);
```

Comment:
This is duplicated code... almost similar and also the logic might be similar to determineScope, but not sure if it pays off now to "improve" this